src/binary/ibc-relayer/setup-prometheus.ts

```diff
@@ -0,0 +1,60 @@
+import http from 'http';
+
+import client from 'prom-client';
+
+import { Logger } from '../create-logger';
+
+let initialized = false;
+
+function getMetrics() {
+  return {
+    pollCounter: new client.Counter({
+      name: 'poll_counter',
+      help: 'Poll counter',
+    }),
+  };
+}
+
+export type Metrics = {
+  pollCounter: client.Counter<string>;
+} | null;
+export function setupPrometheus({
+  enabled,
+  port,
+  logger,
+}: {
+  enabled: boolean;
+  port: number;
+  logger: Logger;
+}): Metrics {
+  if (initialized) {
+    throw new Error(
```

Comment:
This makes sense to find bugs in our code fast.

I was going to suggest to only set `initialized` to true if `enabled` was true, but that means double-registry bugs only surface in production. It is good like this